Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift

```diff
@@ -107,13 +107,15 @@ struct ComboBoxInput: View {
                     .frame(maxWidth: .infinity, alignment: .leading)
                     .foregroundColor(selectedValue != nil ? .primary : .secondary)
                     .accessibilityIdentifier("\(element.label) Value")
-                if selectedValue == nil {
-                    Image(systemName: "list.bullet")
-                        .foregroundColor(.secondary)
-                        .accessibilityIdentifier("\(element.label) Options Button")
-                } else if !inputModel.isRequired && inputModel.isEditable  {
-                    ClearButton { selectedValue = nil }
-                        .accessibilityIdentifier("\(element.label) Clear Button")
+                if inputModel.isEditable {
+                    if selectedValue == nil {
```

Comment:
What if the field does not support `nil` values or the value of `ComboBoxFormInput.noValueOption == .hide`? We wouldn't want to show the `Clear` button in that case, but probably the `list.bullet` image.

Reply:
I had thought so too but Android is placing a clear button in required fields. See [this conversation](https://devtopia.esri.com/runtime/common-toolkit/pull/224#discussion_r1065234) in the test design PR.

Kaushik highlighted that "After Clear" in the design spec implies that required fields can be cleared out:
<img width="317" alt="Screenshot 2023-10-12 at 09 58 56" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/b71a6895-e7a2-4070-b617-a17e9705f6c4">

and Field Maps also places a clear button on a required combo field:
![IMG_0041](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/587b0cc7-f114-476f-b450-16727f5ea988)

As for `ComboBoxFormInput.noValueOption == .hide` the spec is:
<img width="859" alt="Screenshot 2023-10-12 at 10 01 08" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/1b19c7e0-868d-4b77-b888-c03423c6f534"> which is how we have [implemented](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/77f4e5ae12ac8ff24d3c4b7e85105df17f99ccfe/Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift#L230-L232) it.